### PR TITLE
SW-478_fix_multi_byte_filter_names

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: FinSearchUnified/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydrator.php [hunk 1]
```diff
@@ -54,14 +54,14 @@ public function hydrateFacet(SimpleXMLElement $select)
      */
     public function getFormFieldName($name)
     {
-        $escapedName = preg_replace(
-            '/[^\xC3\x96|\xC3\x9C|\xC3\x9F|\xC3\xA4|\xC3\xB6|\xC3\xBC|\x00-\x7F]|[\.\s\x5B]/',
+        $escapedName = mb_ereg_replace(
```
Comment (dblaichinger) [inline FinSearchUnified/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydrator.php:57]: A comment might be helpful here, which chars are getting replaced?

File: FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php [hunk 2]
```diff
@@ -145,6 +145,20 @@ public function facetFilterProvider()
...
+                    'name' => 'TISCHWÄSCHE - Ausführung',
+                    'display' => 'Multibyte - TISCHWÄSCHE - Ausführung',
+                    'select' => 'single'
+                ],
+                'expectedName' => 'TISCHWÄSCHE - Ausführung',
+                'expectedUniqueKey' => 'TISCHWÄSCHE - Ausführung',
+                'expectedAttributeName' => 'product_attribute_TISCHWÄSCHE - Ausführung',
+                'expectedAttributeFormFieldName' => 'TISCHW_SCHE_-_Ausführung',
+                'expectedAttributeLabel' => 'Multibyte - TISCHWÄSCHE - Ausführung',
+                'expectedAttributeMode' => ProductAttributeFacet::MODE_RADIO_LIST_RESULT
```
Comment (dblaichinger) [inline FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php:161]: The structure is different than with the other data provider cases, is that correct?

File: FinSearchUnified/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydrator.php (hunk 1 above)
Comment (TheKeymaster) [reply FinSearchUnified/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydrator.php:57]: Please see the method docblock.

File: FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php (hunk 2 above)
Comment (TheKeymaster) [reply FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php:161]: No I just like to have the names of the parameters for the data provider directly inside of the data provider. This way I can read a data provider without the need of the test method and I can add new cases quicker.

Comment (dblaichinger) [reply FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php:161]: Can you update the other cases than to have the same structure?

**Pull request merged**